osf/models/preprint_provider.py

```diff
@@ -111,6 +113,20 @@ def absolute_api_v2_url(self):
         path = '/preprint_providers/{}/'.format(self._id)
         return api_v2_url(path)
 
+    def save(self):
+        dirty_fields = self.get_dirty_fields()
+        old_id = dirty_fields['_id']
```

Comment:
This would result in `KeyError` if `_id` is not a key. `dirty_fields.get('_id')` is preferable.